File: typescript-client-sdk/src/sdk/models/shared/codatdatacontractsdatasetsbankingtransactionpagedresponselinksmodel.ts
import { SpeakeasyBase, SpeakeasyMetadata } from "../../../internal/utils";
import { CodatDataContractsDatasetsBankingTransactionPagedResponseHrefModel } from "./codatdatacontractsdatasetsbankingtransactionpagedresponsehrefmodel";
import { Expose, Type } from "class-transformer";


export class CodatDataContractsDatasetsBankingTransactionPagedResponseLinksModel extends SpeakeasyBase {
  @SpeakeasyMetadata()
  @Expose({ name: "current" })
  @Type(() => CodatDataContractsDatasetsBankingTransactionPagedResponseHrefModel)
  current?: CodatDataContractsDatasetsBankingTransactionPagedResponseHrefModel;

  @SpeakeasyMetadata()
  @Expose({ name: "next" })
  @Type(() => CodatDataContractsDatasetsBankingTransactionPagedResponseHrefModel)
  next?: CodatDataContractsDatasetsBankingTransactionPagedResponseHrefModel;

  @SpeakeasyMetadata()
  @Expose({ name: "previous" })
  @Type(() => CodatDataContractsDatasetsBankingTransactionPagedResponseHrefModel)
  previous?: CodatDataContractsDatasetsBankingTransactionPagedResponseHrefModel;

  @SpeakeasyMetadata()
  @Expose({ name: "self" })
  @Type(() => CodatDataContractsDatasetsBankingTransactionPagedResponseHrefModel)
  self?: CodatDataContractsDatasetsBankingTransactionPagedResponseHrefModel;
}

File: python-client-sdk/src/sdk/models/shared/codat_datacontracts_datasets_banking_transactionpagedresponselinksmodel.py
import dataclasses
from ..shared import codat_datacontracts_datasets_banking_transactionpagedresponsehrefmodel as shared_codat_datacontracts_datasets_banking_transactionpagedresponsehrefmodel
from dataclasses_json import dataclass_json
from sdk import utils
from typing import Optional


@dataclass_json
@dataclasses.dataclass
class CodatDataContractsDatasetsBankingTransactionPagedResponseLinksModel:
    current: Optional[shared_codat_datacontracts_datasets_banking_transactionpagedresponsehrefmodel.CodatDataContractsDatasetsBankingTransactionPagedResponseHrefModel] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('current') }})
    next: Optional[shared_codat_datacontracts_datasets_banking_transactionpagedresponsehrefmodel.CodatDataContractsDatasetsBankingTransactionPagedResponseHrefModel] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('next') }})
    previous: Optional[shared_codat_datacontracts_datasets_banking_transactionpagedresponsehrefmodel.CodatDataContractsDatasetsBankingTransactionPagedResponseHrefModel] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('previous') }})
    self: Optional[shared_codat_datacontracts_datasets_banking_transactionpagedresponsehrefmodel.CodatDataContractsDatasetsBankingTransactionPagedResponseHrefModel] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('self') }})
    